SimulationRuntime/c/simulation/results/simulation_result_mat.cpp

```diff
@@ -712,47 +725,47 @@ static void generateData_1(DATA *data, threadData_t *threadData, double* &data_1
   const SIMULATION_INFO *sInfo = data->simulationInfo;
   const MODEL_DATA      *mData = data->modelData;
 
-  int offset = 1;
+  int offset = 0;
   long i = 0;
 
   /* calculate number of rows and columns */
-  rows = 2;
-  cols = 1 + mData->nParametersReal +
-             mData->nParametersInteger +
-             mData->nParametersBoolean;
+  cols = mData->nParametersReal +
+         mData->nParametersInteger +
+         mData->nParametersBoolean;
+  if (cols) {
+    rows = 1;
+  } else {
+    rows = 0;
+  }
```

Comment:
@sjoelund What if all of these parameters are skipped due to the variable filter? I guess we have to iterate over all the parameters twice: first to count them and second to copy them to `data_1`.

Reply:
Parameters are not filtered as far as I remember

Reply:
It looks like they get filtered:

https://github.com/OpenModelica/OMCompiler/blob/b610cb9d9d839c81faf23b32c7f67c8c06acacc2/SimulationRuntime/c/simulation/results/simulation_result_mat.cpp#L749-L755

Reply:
Never mind. I have to change that code anyway a bit to add the time-unvarying variables to `data_1`.